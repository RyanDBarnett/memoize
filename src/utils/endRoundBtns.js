import React from 'react';

function endRoundBtns(resetAllQuestions, resetIncorrectQuestions) {
  return (
    <div className="btnContainer">
      <button className="resetAll" onClick={resetAllQuestions}>Reset All Questions</button>
      <button className="resetIncorrect" onClick={resetIncorrectQuestions}>Practice Incorrect Questions</button>
    </div>
  )
}

export default endRoundBtns;