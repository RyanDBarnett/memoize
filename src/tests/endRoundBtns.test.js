import endRoundBtns from '../utils/endRoundBtns';
import { shallow } from "enzyme";

describe('endRoundBtns', () => {
  let wrapper;

  const mockResetAllQuestions = jest.fn();
  const mockResetIncorrectQuestions = jest.fn();

  beforeEach( () => {
    wrapper = shallow(
      endRoundBtns(mockResetAllQuestions, mockResetIncorrectQuestions)
    )
  });

  it("should match the snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should invoke the reset all questions function when reset button is clicked', () => {
    let resetBtn = wrapper.find('.resetAll');
    resetBtn.simulate('click');
    expect(mockResetAllQuestions).toHaveBeenCalled()
  });

  it('should invoke the reset incorrect questions function when reset incorrect button is clicked', () => {
    let resetIncorrectBtn = wrapper.find('.resetIncorrect');
    resetIncorrectBtn.simulate('click');
    expect(mockResetIncorrectQuestions).toHaveBeenCalled()
  });
});