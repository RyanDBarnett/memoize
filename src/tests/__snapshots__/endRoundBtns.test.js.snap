// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`endRoundBtns should match the snapshot 1`] = `
<div
  className="btnContainer"
>
  <button
    className="resetAll"
    onClick={[MockFunction]}
  >
    Reset All Questions
  </button>
  <button
    className="resetIncorrect"
    onClick={[MockFunction]}
  >
    Practice Incorrect Questions
  </button>
</div>
`;
